tempodb/search/backend_search_block_test.go

```diff
@@ -28,15 +28,16 @@ func genSearchData(traceID []byte, i int) [][]byte {
 }
 
 func newBackendSearchBlockWithTraces(t testing.TB, traceCount int, enc backend.Encoding, pageSizeBytes int) *BackendSearchBlock {
-	id := []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15} // 16-byte ids required
-
 	f, err := os.OpenFile(path.Join(t.TempDir(), "searchdata"), os.O_CREATE|os.O_RDWR, 0644)
 	require.NoError(t, err)
 
 	b1, err := NewStreamingSearchBlockForFile(f)
 	require.NoError(t, err)
+
 	for i := 0; i < traceCount; i++ {
-		assert.NoError(t, b1.Append(context.Background(), id, genSearchData(id, i)))
+		id := make([]byte, 16)
+		binary.LittleEndian.PutUint32(id, uint32(i))
```

Comment:
Making a note to self: Need to add a new test for deduping WAL segments if the traceIDs are going to become unique here

Reply:
This is done here - https://github.com/grafana/tempo/pull/968